### DISCUSSION TITLE
defining board type

### DISCUSSION THREAD

**Discussion by ithinkido** [dbuezas/lgt8fx#197] [Q&A] [answered]
What would be the correct way to define the LGT8F 328P board type similar to how this would work for an AVR 328P ?

eg.

```#ifdef __AVR_ATmega328P__  ```

**Comment by LaZsolt** [comment, marked as answer]
https://github.com/dbuezas/lgt8fx/wiki/Identifying-MCU-and-board-variants-using-preprocessor

**Comment by ithinkido** [comment]
Perfect ! Thank you for the link.

**Comment by ithinkido** [comment]
Is it possible that there is some sort of overlap between the (__AVR__) and (__LGT8FX8P__) board defnitions. I am getting false positives when using ifdef statements to seperate code for the oard types.

An example of code  to demonstrate the false positives would be the following :  

```
#ifdef __AVR__
#error AVR 
#endif

#ifdef __LGT8FX8P__
#error LGT
#endif



void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}
```
![image](https://user-images.githubusercontent.com/52834821/151703918-b27641fd-9b13-4a3e-b966-7317d3ca93f5.png)


Is there something I could do better to use the board definitions to separate code out dependent on the board type?

**Reply by SuperUserNameMan** [reply]
IIRC, it's normal `__AVR__` is also defined on LGT board, because LGT boards are also AVR architecture.

I think you should test that way instead :
```C
#ifdef __LGT8F__
#error LGT board
#else
#error Non LGT board
#endif
```

**Reply by ithinkido** [reply]
A bit confusing, but in the end, if it works, it works. Thank you for the answer
